### PR TITLE
Lightning navigation

### PR DESCRIPTION
**Pull request by andrewgryan**
A simple file name parsing navigator, ideal for S3 navigation no need to touch 57,000 files to populate a dropdown

### CODE REVIEW

**Comment by andrewgryan** [comment]
A Navigator dedicated to EIDA50 might be more appropriate here, the existing Locator/Loader feels a bit clunky and hacky

**Comment by andrewgryan** [comment]
@twsearle can you take a look at this naive attempt to implement lightning navigation

File: forest/earth_networks.py [hunk 1]
```diff
@@ -77,8 +67,39 @@ def add_figure(self, figure):
...
+    def valid_times(self, pattern, variable, initial_time):
+        return self._valid_times
+
+    def pressures(self, pattern, variable, initial_time):
+        return []
+
+
 class Loader(object):
     def __init__(self, paths):
+        print("Loader", paths)
```
Comment (twsearle) [inline forest/earth_networks.py]: delete

File: forest/satellite.py [hunk 2]
```diff
@@ -98,10 +103,16 @@ def find_index(times, time, length):
...
+            print(msg)
             raise IndexNotFound(msg)
         return np.ma.argmax(valid_times)
 
     @staticmethod
     def parse_date(path):
         groups = re.search(r"([0-9]{8})\.nc", path)
-        return dt.datetime.strptime(groups[1], "%Y%m%d")
+        if groups is None:
+            # *20191211T0000Z.nc
```
Comment (twsearle) [inline forest/satellite.py]: stray comment.

File: forest/satellite.py [hunk 3]
```diff
@@ -42,8 +42,10 @@ def load_image(self, path, itime):
         fraction = 0.25
         lons, lats, values = coarsify(
                 lons, lats, values, fraction)
-        return geo.stretch_image(
+        data = geo.stretch_image(
                 lons, lats, values)
+        print(data["image"][0])
```
Comment (twsearle) [inline forest/satellite.py]: stray print statement

File: forest/satellite.py [hunk 4]
```diff
@@ -58,6 +60,7 @@ def find(self, date):
         ipath = self.find_file_index(paths, date)
         path = paths[ipath]
         time_axis = self.load_time_axis(path)
+        print(path, time_axis, date)
```
Comment (twsearle) [inline forest/satellite.py]: stray print statement

File: forest/earth_networks.py [hunk 5]
```diff
@@ -77,8 +67,39 @@ def add_figure(self, figure):
         return renderer
 
 
+class Navigator:
+    def __init__(self, paths):
+        print("Navigator", paths)
```
Comment (twsearle) [inline forest/earth_networks.py]: stray print statement

File: forest/satellite.py (hunk 2 above)
Comment (andrewgryan) [reply forest/satellite.py]: This comment could be extended to explain itself

**Pull request merged**